webhook_router/app.py

```diff
@@ -2,17 +2,105 @@
 #  See LICENSE file for licensing details.
 
 """Flask application which receives GitHub webhooks and logs those."""
-import json
 import logging
 
+import yaml
 from flask import Flask, request
 
+from webhook_router.mq import add_job_to_queue
 from webhook_router.validation import verify_signature
+from webhook_router.webhook import Job, JobStatus
+from webhook_router.webhook.label_translation import (
+    FlavorLabelsMapping,
+    InvalidLabelCombinationError,
+    labels_to_flavor,
+    to_labels_flavor_mapping,
+)
+from webhook_router.webhook.parse import ParseError, webhook_to_job
 
+SUPPORTED_GITHUB_EVENT = "workflow_job"
+GITHUB_EVENT_HEADER = "X-Github-Event"
 WEBHOOK_SIGNATURE_HEADER = "X-Hub-Signature-256"
 
-app = Flask(__name__)
-app.config.from_prefixed_env()
+app = Flask(__name__.split(".", maxsplit=1)[0])
+
+
+class ConfigError(Exception):
+    """Raised when a configuration error occurs."""
+
+
+def config_app(flask_app: Flask) -> None:
+    """Configure the application.
+
+    Args:
+        flask_app: The Flask application to configure.
+    """
+    flask_app.config.from_prefixed_env()
+    flavor_labels_mapping = _parse_flavors_config(flask_app.config.get("FLAVOURS", ""))
```

Comment:
discussed internally: don't pass a load function, because errors on it will be ignored (see https://flask.palletsprojects.com/en/2.3.x/api/#flask.Config.from_prefixed_env)